### PR TITLE
Bump minimum php to 7.1

### PR DESCRIPTION
**Pull request by Xerkus**
This PR bumps minimum php version to 7.1 and enables strict types

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,5 +18,20 @@ request and responses, and provides capabilities around:
...
+
+Run the following to install this library:
+
+```bash
+$ composer require zendframework/zend-router
+```
+
+## Documentation
+
+Documentation is [in the doc tree](docs/book/), and can be compiled using [mkdocs](http://www.mkdocs.org):
```
Comment (froschdesign) [inline README.md]: Why should that be interesting for the end user? The user wants to read the documentation and not create it himself.

Comment (Xerkus) [reply README.md]: I applied README template: https://github.com/zendframework/maintainers/blob/master/template/README.md

File: README.md [hunk 2]
```diff
@@ -18,5 +18,20 @@ request and responses, and provides capabilities around:
 Additionally, it supports combinations of different route types in tree
 structures, allowing for fast, b-tree lookups.
 
-- File issues at https://github.com/zendframework/zend-router/issues
-- Documentation is at https://docs.zendframework.com/zend-router
+## Installation
```
Comment (froschdesign) [inline README.md]: My suggestion for this changes: before we change the readme we should define the content and structure for all components.
I know you have copied the content, but the current state does not make sense!

File: README.md (hunk 1 above)
Comment (froschdesign) [reply README.md]: Ah, okay. Then we should change the template. Thanks for the very quick response. 👍

Comment (froschdesign) [reply README.md]: I created a PR: https://github.com/zendframework/maintainers/pull/40

**Comment by Xerkus** [comment]
Rebased onto current develop

**Pull request merged**